flight/Libraries/math/filter.c

```diff
@@ -0,0 +1,156 @@
+/**
+ ******************************************************************************
+ * @addtogroup TauLabsLibraries Tau Labs Libraries
+ * @{
+ * @addtogroup TauLabsMath Filtering support libraries
+ * @{
+ *
+ * @file       filter.c
+ * @author     dRonin, http://dronin.org, Copyright (C) 2017
+ * @brief      Signal filtering algorithms
+ *
+ * @see        The GNU Public License (GPL) Version 3
+ *
+ *****************************************************************************/
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, see <http://www.gnu.org/licenses/>
+ */
+
+#include <stdio.h>
+#include <stdint.h>
+#include <string.h>
+#include <math.h>
+#include "pios_heap.h"
+#include "filter.h"
+
+const float filter_butterworth_factors[16] = {
```

Comment:
All of the things not accessed directly by callers-- this, construct_single_biquad, construct_biquads.. should be declared 'static'